packages/node-resolve/src/resolveId.js

```diff
@@ -0,0 +1,130 @@
+import fs from 'fs';
+import path from 'path';
+import { promisify } from 'util';
+
+import resolve from 'resolve';
+
+import { getPackageName } from './util';
```

Comment:
There is a cyclic dependency between this file and the `util` file. While this does not have to be an issue, cyclic dependencies tend to cause confusing errors down the road and I would prefer to avoid them. My suggestion would be to extract `resolveImportSpecifiers` from the `util` file and either move it into its own file or into this one.

Reply:
Removed the circular dependency.